c/pkg/objects/vm_disk.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2019 Red Hat, Inc.
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package objects
+
+import (
+	"github.com/virtblocks/virtblocks/go/pkg/vm"
+	"sync"
+)
+
+var vmDiskObjectsLock sync.RWMutex
+var vmDiskObjects = make([]*vm.Disk, 1)
+
+func VmDiskAdd(disk *vm.Disk) uint {
+	vmDiskObjectsLock.Lock()
```

Comment:
I wonder if this is a good pattern for everything. It might make sense to just copy things as a whole a little bit more.

Reply:
For something simple like `vm.Disk` that might be fine, but as soon as you get to something just a little bit more complicated (such as `vm.Description`) then copying becomes fairly involved...

Also you'd have to do that conversion every time you cross the language barrier, whereas with the current approach you're only really passing integers around and for the most part dealing with actual Go object in the C bindings, which is quite a bit nicer I think.

So unless I'm missing something - always a very real possibility! - I think the current approach, or some variation of it, is what we want.

Reply:
:+1: